addons/minio/2020-01-25T02-50-51Z/install.sh

```diff
@@ -25,16 +35,84 @@ function minio() {
 function minio_creds() {
     local src="$1"
     local dst="$2"
- 
-    if kubernetes_resource_exists ${MINIO_NAMESPACE} secret minio-credentials; then
-        return 0
-    fi
 
     local MINIO_ACCESS_KEY=kurl
-    local MINIO_SECRET_KEY=$(< /dev/urandom tr -dc A-Za-z0-9 | head -c9)
+    local MINIO_SECRET_KEY=$(kubernetes_secret_value minio minio-credentials MINIO_SECRET_KEY)
+
+    if [ -z "$MINIO_SECRET_KEY" ]; then
+        MINIO_SECRET_KEY=$(< /dev/urandom tr -dc A-Za-z0-9 | head -c9)
+    fi
 
     render_yaml_file "$src/tmpl-creds-secret.yaml" > "$dst/creds-secret.yaml"
     insert_resources "$dst/kustomization.yaml" creds-secret.yaml
+
+    if [ -n "$MINIO_HOSTPATH" ]; then
+        # in the case of using a "hostPath", minio will generate a config in a ".minio.sys" directory
+        # that is used to control access to that hostPath. this can be a problem when that hostpath
+        # is a directory in a shared file system (an NFS mount for example) and the installer is being
+        # run on a fresh instance, because new credentials will be generated and minio won't be able access the old data,
+        # so we make sure that the config is regenerated from the current minio credentials.
+
+        # initialize some common variables
+        local MINIO_CONFIG_PATH="$MINIO_HOSTPATH/.minio.sys/config"
+        local KURL_DIR="$MINIO_HOSTPATH/.kurl"
+        local MINIO_KEYS_SHA_FILE="$KURL_DIR/minio-keys-sha.txt"
+        local NEW_MINIO_KEYS_SHA=$(echo -n "$MINIO_ACCESS_KEY,$MINIO_SECRET_KEY" | sha256sum)
+
+        if should_reset_minio_config; then
+            reset_minio_config
+        fi
+        if [ ! -d "$MINIO_CONFIG_PATH" ]; then
+            kubernetes_scale_down ${MINIO_NAMESPACE} deployment minio
+        fi
+        write_minio_keys_sha_file
+    fi
+}
+
+function should_reset_minio_config() {
+    if [ -z "$MINIO_HOSTPATH" ]; then
+        return 1
+    fi
+
+    if [ ! -d "$MINIO_CONFIG_PATH" ]; then
+        return 1
+    fi
+
+    if ! kubernetes_resource_exists ${MINIO_NAMESPACE} secret minio-credentials; then
+        return 0
+    fi
+
+    if [ ! -f "$MINIO_KEYS_SHA_FILE" ]; then
+        return 0
+    fi
+
+    local EXISTING_MINIO_KEYS_SHA=$(cat $MINIO_KEYS_SHA_FILE)
+    if [ "$NEW_MINIO_KEYS_SHA" == "$EXISTING_MINIO_KEYS_SHA" ]; then
+        return 1
+    fi
+
+    return 0
+}
+
+function reset_minio_config() {
+    if [ ! -d "$MINIO_CONFIG_PATH" ]; then
+        return 0
+    fi
+
+    printf "\n${YELLOW}The $MINIO_HOSTPATH directory will be re-configured to be used by this minio instance. Would you like to continue?${NC} "
+    if ! confirmN "-t 120"; then
+        bail "\nWill not re-configure $MINIO_HOSTPATH."
+    fi
+
+    rm -rf "$MINIO_CONFIG_PATH"
```

Comment:
Since this will render minio data useless, how about `rm -rf $KURL_DIR` and then make that warning more dire and color it red?